plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -140,15 +151,28 @@ public static TableStatistics makeTableStatistics(
                 .map(column -> Maps.immutableEntry(column.fieldId(), column.type()))
                 .collect(toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
 
+        Domain pathDomain = getPathDomain(effectivePredicate);
+        Domain fileModifiedTimeDomain = getFileModifiedTimePathDomain(effectivePredicate);
         TableScan tableScan = icebergTable.newScan()
-                // Table enforced constraint may include eg $path column predicate which is not handled by Iceberg library TODO apply $path and $file_modified_time filters here
                 .filter(toIcebergExpression(effectivePredicate.filter((column, domain) -> !isMetadataColumnId(column.getId()))))
                 .useSnapshot(snapshotId)
                 .includeColumnStats();
 
         IcebergStatistics.Builder icebergStatisticsBuilder = new IcebergStatistics.Builder(columns, typeManager);
         try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
-            fileScanTasks.forEach(fileScanTask -> icebergStatisticsBuilder.acceptDataFile(fileScanTask.file(), fileScanTask.spec()));
+            fileScanTasks.forEach(fileScanTask -> {
+                if (!pathDomain.includesNullableValue(utf8Slice(fileScanTask.file().path().toString()))) {
```

Comment:
optional: We could also add another test in `TestIcebergFileOperations` which would showcase that calling `SHOW STATS` with a filter on `$path` or `modified_time` domain shrinks very much the amount of data files to be taken into account for processing.

Reply:
@findinpath @mosabua Thanks for the attention

Reply:
I tried write the tests for `path` and `file_modified_time` column with `SHOW STATS`, but found that the files access almost the same. for the `file_modified_time` there are even more access with `INPUT_FILE_LAST_MODIFIED` type file operation.